src/opencv_example.cc

```diff
@@ -0,0 +1,128 @@
+// out-of-the-box depending on your system and camera used. See the README for
+// more information.
+
+// Standard C++ library includes
+#include <chrono>
+#include <string>
+#include <vector>
+
+// Regular OpenCV libraries, these should be installed on your system
+#include "opencv2/core.hpp"
+#include "opencv2/highgui/highgui.hpp"
+#include "opencv2/videoio.hpp"
+
+// The Plumerai People Detection library. If you do not have access to this
+// library and would like to evaluate it, then contact us at hello@plumerai.com.
+// See docs.plumerai.com for information how to use it.
+#include "plumerai/people_detection.h"
+
+// Input camera settings (change as needed)
+constexpr auto width = 1280;
+constexpr auto height = 720;
+constexpr auto camera_id = 0;  // 0 == the first camera, 1 == the second, etc.
+
+// The following line sets the capture API to Video4Linux. Change this as needed
+// on your system. See the OpenCV website for a list of all options:
+// https://docs.opencv.org/4.6.0/d4/d15/group__videoio__flags__base.html
+constexpr auto camera_capture_api = cv::CAP_V4L;
+
+// The following line sets the camera format to Motion-JPEG. If your camera does
+// not support this, then it could be changed to another format.
+const auto camera_capture_format = cv::VideoWriter::fourcc('M', 'J', 'P', 'G');
+
+// These are settings to draw bounding boxes on screen. They do not need to be
+// modified. A bounding-box will use any of the following 19 colours. If there
+// are more than 19 boxes found, it will re-use the same colour.
+constexpr int num_colours = 19;
+const int colours[num_colours][3] = {
+    {230, 25, 75},  {60, 180, 75},   {255, 225, 25}, {0, 130, 200},
+    {245, 130, 48}, {145, 30, 180},  {70, 240, 240}, {240, 50, 230},
+    {210, 245, 60}, {250, 190, 212}, {0, 128, 128},  {220, 190, 255},
+    {170, 110, 40}, {255, 250, 200}, {128, 0, 0},    {170, 255, 195},
+    {128, 128, 0},  {255, 215, 180}, {0, 0, 128}};
+
+int main() {
+  // Set-up the Plumerai People Detection algorithm
+  auto ppd = plumerai::PeopleDetection(height, width);
+
+  // Set-up the output window
+  const auto window_text = "Plumerai People Detection";
+  cv::namedWindow(window_text, cv::WINDOW_AUTOSIZE);
+
+  // Set-up the camera, see the above constants for the settings
+  cv::VideoCapture camera(camera_id, camera_capture_api);
+  camera.set(cv::CAP_PROP_FRAME_WIDTH, width);
+  camera.set(cv::CAP_PROP_FRAME_HEIGHT, height);
+  camera.set(cv::CAP_PROP_FOURCC, camera_capture_format);
+  if (!camera.isOpened()) {
+    // If this is printed the camera might not be attached or it might not
+    // support the settings set above (width & height, capture API or format)
+    printf("Error: Could not open camera %d, verify the settings\n", camera_id);
+    return -1;
+  }
+
+  // Run the demo in an infinite loop
+  cv::Mat cv_image;
+  while (true) {
+    // Retrieve an image from the camera
+    camera.read(cv_image);
```

Comment:
```suggestion
    if (camera.read(cv_image) == false) {
      printf("Error: could not read frame.\n");
      return -1;
    }
```
This could happen when the webcam is disconnected, or when the user replaces line 58 with `VideoCapture camera("videofile.mp4");` and the end of the video is reached.